internal/xds/translator/cluster.go

```diff
@@ -461,45 +465,90 @@ func buildXdsClusterLoadAssignment(clusterName string, destSettings []*ir.Destin
 			}
 		}
 
-		for _, irEp := range ds.Endpoints {
-			healthStatus := corev3.HealthStatus_UNKNOWN
-			if irEp.Draining {
-				healthStatus = corev3.HealthStatus_DRAINING
-			}
-			lbEndpoint := &endpointv3.LbEndpoint{
-				Metadata: metadata,
-				HostIdentifier: &endpointv3.LbEndpoint_Endpoint{
-					Endpoint: &endpointv3.Endpoint{
-						Address: buildAddress(irEp),
-					},
+		if ds.ZoneAwareRoutingEnabled {
```

Comment:
To help people come later understand why buildZonalLocalities is chosen over buildWeightedLocalities, can we add a few lines to explain the rations here?

It would be helpful to add links to the existing comments and issues on the problems of using `WeightedCluster` 
https://github.com/envoyproxy/gateway/issues/5307#issuecomment-2688767482

Reply:
this section isnt related to weighted cluster though

Reply:
Was planning to add that with the zone discovery PR along with some docs but I'll add some code comments

Reply:
Updated